Error: Can't resolve 'aws-sdk' in ... epsagon/dist'

Hi,

I have a lambda function that returns a string. We are using serverless framework for build and distribution. We are getting below error when we run serverless webpack

ERROR in ./node_modules/epsagon/dist/bundle.js
Module not found: Error: Can't resolve 'aws-sdk' in '/Users/Documents/02. Projects/aws/newproj/packages/backend/dt_aws/node_modules/epsagon/dist'
 @ ./node_modules/epsagon/dist/bundle.js 1:428-446
 @ ./epsogan.ts
 @ ./func/hellofunc.ts
 @ multi ./func/hellofunc.ts

If I understand above error correctly webpack is trying to find aws-sdk to bundle and failing because it cannot find it. If this is true then do we need to add aws-sdk to dev dependency? 

Thanks,

Kal

--- a/src/events/aws_sdk.js
+++ b/src/events/aws_sdk.js
@@ -1,9 +1,9 @@
 /**
  * @fileoverview Handlers for the aws-sdk js library instrumantation.
  */
-const AWS = require('aws-sdk'); // eslint-disable-line import/no-extraneous-dependencies
 const md5 = require('md5');
 const uuid4 = require('uuid4');
+const tryRequire = require('try-require');
 const shimmer = require('shimmer');
 JSON.sortify = require('json.sortify');
 const utils = require('../utils.js');
@@ -13,6 +13,8 @@ const eventInterface = require('../event.js');
 const errorCode = require('../proto/error_code_pb.js');
 const { STEP_ID_NAME } = require('../consts.js');
 const resourceUtils = require('../resource_utils/sqs_utils.js');
+
+const AWS = tryRequire('aws-sdk');
 
 const s3EventCreator = {
     /**
@@ -748,7 +750,9 @@ module.exports = {
      * Initializes the aws-sdk tracer
      */
     init() {
-        shimmer.wrap(AWS.Request.prototype, 'send', AWSSDKWrapper);
-        shimmer.wrap(AWS.Request.prototype, 'promise', AWSSDKWrapper);
+        if (AWS) {
+            shimmer.wrap(AWS.Request.prototype, 'send', AWSSDKWrapper);
+            shimmer.wrap(AWS.Request.prototype, 'promise', AWSSDKWrapper);
+        }
     },
 };

